src/components/textfieldControl/js/TextfieldSlider.js

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2013-2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /********************
+     * Textfield Slider *
+     ********************/
+
+    fluid.defaults("fluid.textfieldSlider", {
+        gradeNames: ["fluid.viewComponent"],
+        components: {
+            textfield: {
+                type: "fluid.textfield",
+                container: "{that}.dom.textfield",
+                options: {
+                    gradeNames: ["fluid.textfield.rangeController"],
```

Comment:
What about directly using `fluid.textfield.rangeController` at line 25? The same comment applies to the implementation of `fluid.textfieldStepper`.